src/kernel/mpipeline.ml

```diff
@@ -156,18 +236,41 @@ let process
           raw_source, Some ast
     )) in
   let reader = timed_lazy reader_time (lazy (
-      let lazy source = source in
+      let lazy ((_, pp_result) as source) = source in
       let config = Mconfig.normalize config in
       Mocaml.setup_reader_config config;
-      let result = Mreader.parse ?for_completion config source in
-      result, config
+      let { Reader_with_cache.output; cache_was_hit } =
+      let cache_disabling =
+        match (config.merlin.use_ppx_cache, pp_result) with
+        | false, _ -> Some "configuration"
+        | true, Some _ -> Some "source preprocessor usage"
```

Comment:
Couldn't we use the preprocessed parse tree as a fingerprint when the source is preprocessed and have the cache working ?

Reply:
Btw, I've just added a small comment about this in the code.